### PR TITLE
[WIP] Switch to JDK 14 for releasing and CI build

### PR DESCRIPTION
**Pull request by minwoox**

### CODE REVIEW

File: .appveyor.build.sh [hunk 1]
```diff
@@ -5,10 +5,10 @@ JRE8_URL='https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jd
 JRE8_VERSION='AdoptOpenJDK-8u242b08'
 JRE11_URL='https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.6%2B10/OpenJDK11U-jre_x64_linux_hotspot_11.0.6_10.tar.gz'
 JRE11_VERSION='AdoptOpenJDK-11.0.6_10'
-JRE13_URL='https://github.com/AdoptOpenJDK/openjdk13-binaries/releases/download/jdk-13.0.2%2B8/OpenJDK13U-jdk_x64_linux_hotspot_13.0.2_8.tar.gz'
-JRE13_VERSION='AdoptOpenJDK-13.0.2_8'
-BUILD_JDK_URL="$JRE13_URL"
-BUILD_JDK_VERSION="$JRE13_VERSION"
+JRE14_URL='https://github.com/AdoptOpenJDK/openjdk14-binaries/releases/download/jdk-14%2B36/OpenJDK14U-jre_x64_linux_hotspot_14_36.tar.gz'
```
Comment (anuraaga) [inline .appveyor.build.sh]: This URL looks like it's for a JRE not a JDK so won't be able to build

Comment (minwoox) [reply .appveyor.build.sh]: Ah oops, let me fix this. Thank you!

**Comment by minwoox** [comment]
It seems like we need to wait for Proguard gradle 6.3.0. 🤔

**Comment by anuraaga** [comment]
Proguard was a problem last time too.

**Comment by anuraaga** [comment]
(lately I've been having a lot of problems clicking buttons on mobile accidentally 😭)

**Comment by adriancole** [comment]
will Proguard gradle 7 work?

PS JDK 15 is out, though not on all flavors yet

**Comment by ikhoon** [comment]
The Java 14 issue seems to be resolved with Proguard 7.
https://github.com/Guardsquare/proguard/releases/tag/v7.0.0
@minwoox is busy these days to prepare lessons to teach Armeria and its internal architecture to LINE employees. Let me test this PR with Proguard 7 😉

**Comment by minwoox** [comment]
Superseded by #3073

**Pull request closed without merging**